pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -738,8 +1009,52 @@ func (pl *DynamicResources) Unreserve(ctx context.Context, cs fwk.CycleState, po
 	for index, claim := range state.claims {
 		// If allocation was in-flight, then it's not anymore and we need to revert the
 		// claim object in the assume cache to what it was before.
-		if deleted := pl.draManager.ResourceClaims().RemoveClaimPendingAllocation(state.claims[index].UID); deleted {
-			pl.draManager.ResourceClaims().AssumedClaimRestore(claim.Namespace, claim.Name)
+
+		uid := state.claims[index].UID
+		if pl.enableExtendedResource {
+			if uid == state.extendedResourceClaimAPIUID {
+				uid = state.extendedResourceClaimTplUID
+			}
+		}
+		if deleted := pl.draManager.ResourceClaims().RemoveClaimPendingAllocation(uid); deleted {
+			if !pl.enableExtendedResource || claim.Name != "" {
+				pl.draManager.ResourceClaims().AssumedClaimRestore(claim.Namespace, claim.Name)
+			}
+		}
+
+		if pl.enableExtendedResource {
+			if (claim.UID == state.extendedResourceClaimAPIUID || claim.UID == state.extendedResourceClaimTplUID) && claim.Name != "" {
```

Comment:
Instead of checking UIDs I find it simpler to track a `state.extendedResourceClaim *resourceapi.ResourceClaim` and compare against that.